src/openrct2/ride/Ride.cpp

```diff
@@ -5340,11 +5340,11 @@ void Ride::SetNameToDefault()
 /**
  * This will return the name of the ride, as seen in the New Ride window.
  */
-RideNaming get_ride_naming(const uint8_t rideType, rct_ride_entry* rideEntry)
+RideNaming get_ride_naming(const RideTypeDescriptor& rideType, rct_ride_entry* rideEntry)
```

Comment:
This is a good refactor by itself, but it would be better to split off and merge first, to get a clearer diff.